sourcecode/apis/contentauthor/tests/Unit/Models/H5PContentTest.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+namespace Tests\Unit\Models;
+
+use App\H5PContent;
+use App\H5PContentsMetadata;
+use App\H5PLibrary;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+use Illuminate\Http\Request;
+use Tests\TestCase;
+use Tests\Traits\WithFaker;
+
+class H5PContentTest extends TestCase
+{
+    use RefreshDatabase;
+    use WithFaker;
+
+    public function setUp(): void
+    {
+        parent::setUp();
+        $this->refreshDatabase();
+    }
+
+    /**
+     * @covers \App\H5PContent::requestShouldBecomeNewVersion
+    */
+    public function testRequestShouldBecomeNewVersion(): void
+    {
+        /** @var H5PLibrary $library */
+        $library = H5PLibrary::factory()->create([
+            'name' => 'H5P.UnitTest',
+            'major_version' => 4,
+            'minor_version' => 2,
+            'patch_version' => 0,
+        ]);
+        H5PLibrary::factory()->create([
+            'name' => 'H5P.UnitTest',
+            'major_version' => 4,
+            'minor_version' => 3,
+            'patch_version' => 0,
+        ]);
+        /** @var H5PContent $content */
+        $content = H5PContent::factory()->create([
+            'title' => $this->faker->words(3, true),
+            'is_draft' => true,
+            'library_id' => $library->id,
+        ]);
+        H5PContentsMetadata::factory()->create([
+            'content_id' => $content->id,
+        ]);
+
+        // Resource is draft
```

Comment:
Please make each of these assertions their own test cases. If one of these assertions fail, we can't see if the following ones would have succeeded or not.